.github/workflows/codeql-analysis.yml

```
@@ -42,6 +42,14 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3.6.0
 
+      # Workaround for go1.21 compatibility. See Issue #740.
+      - name: setup-go
```

Just as an FYI:

There is an issue open https://github.com/github/codeql-action/issues/1842

It seems that the CodeQL action relies on the Go version on the runner by default (if you don't use `setup-go`). Therefore, I think this is a reasonable fix.

We should be able to remove it when the GHA runners have updated to Go 1.21+

nit: Make the note about #740 into a TODO comment.
```suggestion
      # TODO(#740): Workaround for go1.21 compatibility. Remove when GHA runners have Go 1.21+
      - name: setup-go
```